pkg/decision/interface.go

```diff
@@ -39,3 +39,9 @@ type ExperimentService interface {
 type FeatureService interface {
 	GetDecision(decisionContext FeatureDecisionContext, userContext entities.UserContext) (FeatureDecision, error)
 }
+
+// UserProfileService is used to save and retrieve past bucketing decisions for users
+type UserProfileService interface {
+	Lookup(string) (UserProfile, error)
```

Comment:
Do we need `Lookup` to return an error? The SDK internals don't have to care whether there was an error - that seems more like a concern of the implementation. Whoever implements this could track and handle errors on their end. Do they really need to route an error through the SDK logger like that?

From the SDK's perspective, it just needs a user profile for a given user id. We could require that a `UserProfile` is always returned, and say that when there is no saved decision for that user id (for whatever reason, error or not), then the implementation should return a `UserProfile` containing an empty `ExperimentBucketMap`.

Reply:
This is a good point. I'll remove the errors since we don't actually care about them.